### PR TITLE
[FIX] product_supplierinfo_for_customer_sale: TypeError: unorderable types: int() >= NoneType()

### PR DESCRIPTION
**Pull request by hugho-ad**
image

### CODE REVIEW

File: product_supplierinfo_for_customer_sale/models/product_product.py [hunk 1]
```diff
@@ -13,6 +13,8 @@ class ProductProduct(models.Model):
     def name_search(self, name='', args=None, operator='ilike', limit=80):
         res = super(ProductProduct, self).name_search(
             name, args=args, operator=operator, limit=limit)
+        if not limit or len(res) >= limit:
+            limit = (limit - len(res)) if limit else False
```
Comment (pedrobaeza) [inline product_supplierinfo_for_customer_sale/models/product_product.py:17]:
```suggestion
            limit = (limit - len(res)) if limit else sys.maxsize
```
importing `sys` previously

File: product_supplierinfo_for_customer_sale/models/product_product.py [hunk 2]
```diff
@@ -13,6 +13,8 @@ class ProductProduct(models.Model):
...
         if (not name or not self._context.get('partner_id') or
```
Comment (pedrobaeza) [inline product_supplierinfo_for_customer_sale/models/product_product.py]: or alternatively
```suggestion
        if (not name and limit or not self._context.get('partner_id') or
```

Comment (hugho-ad) [reply product_supplierinfo_for_customer_sale/models/product_product.py]: Thanks!, Done!

**Review by ValentinVinagre — APPROVED** [review]
LGTM

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**